crates/rpc/src/handlers/block.rs

```diff
@@ -86,28 +80,54 @@ pub async fn get_beacon_block_from_id(
         )))
 }
 
-/// Called by `/eth/v2/beacon/{block_id}/attestations` to get block attestations
-pub async fn get_block_attestations(block_id: ID, db: ReamDB) -> Result<impl Reply, Rejection> {
-    let beacon_block = get_beacon_block_from_id(block_id, &db).await?;
+/// Called by `/genesis` to get the Genesis Config of Beacon Chain.
+#[get("/beacon/genesis")]
+pub async fn get_genesis(db: web::Data<ReamDB>) -> actix_web::Result<impl Responder> {
+    Ok(HttpResponse::Ok())
+}
+
+/// Called by `/eth/v2/beacon/blocks/{block_id}/attestations` to get block attestations
+#[get("/beacon/blocks/{block_id}/attestations")]
+pub async fn get_block_attestations(
+    db: web::Data<ReamDB>,
+    block_id: web::Path<String>,
+) -> actix_web::Result<impl Responder> {
+    let block_id = ID::from_str(&block_id.into_inner()).map_err(error::ErrorBadRequest)?;
+
+    let beacon_block = get_beacon_block_from_id(block_id, &db)
+        .await
+        .map_err(error::ErrorInternalServerError)?;
 
-    Ok(with_status(
-        BeaconVersionedResponse::json(beacon_block.message.body.attestations),
-        StatusCode::OK,
-    ))
+    Ok(HttpResponse::Ok().json(beacon_block.message.body.attestations))
 }
 
 /// Called by `/blocks/<block_id>/root` to get the Tree hash of the Block.
-pub async fn get_block_root(block_id: ID, db: ReamDB) -> Result<impl Reply, Rejection> {
-    let block_root = get_block_root_from_id(block_id, &db).await?;
-    Ok(with_status(
-        BeaconResponse::json(RootResponse { root: block_root }),
-        StatusCode::OK,
-    ))
+#[get("/beacon/blocks/{block_id}/root")]
+pub async fn get_block_root(
+    db: web::Data<ReamDB>,
+    block_id: web::Path<String>,
+) -> actix_web::Result<impl Responder> {
+    let block_id = ID::from_str(&block_id.into_inner()).map_err(error::ErrorBadRequest)?;
```

Comment:
```suggestion
    db: web::Data<ReamDB>,
    block_id: web::Path<ID>,
) -> actix_web::Result<impl Responder> {
```

Apply this feedback to all where applicable